kratos/modeler/combine_model_part_modeler.cpp

```diff
@@ -0,0 +1,413 @@
+// System includes
+#include <string>
+#include <vector>
+#include <sstream>
+
+// External includes
+
+// Project includes
+#include "includes/define.h"
+#include "modeler/combine_model_part_modeler.h"
+
+namespace Kratos
+{
+
+CombineModelPartModeler::CombineModelPartModeler(
+    Model& rModel,
+    Parameters ModelerParameters
+    ) : Modeler(rModel, ModelerParameters),
+        mpModel(&rModel),
+        mParameters(ModelerParameters)
+{
+    mParameters.ValidateAndAssignDefaults(GetDefaultParameters());
+}
+
+/***********************************************************************************/
+/***********************************************************************************/
+
+Modeler::Pointer CombineModelPartModeler::Create(
+    Model& rModel,
+    const Parameters ModelParameters
+    ) const
+{
+    return Kratos::make_shared<CombineModelPartModeler>(rModel, ModelParameters);
+}
+
+/***********************************************************************************/
+/***********************************************************************************/
+
+void CombineModelPartModeler::SetupModelPart()
+{
+    KRATOS_TRY;
+    const auto& r_new_model_part_name = mParameters["combined_model_part_name"].GetString();
+
+    auto& r_combined_model_part = mpModel->HasModelPart(r_new_model_part_name) ? 
+        mpModel->GetModelPart(r_new_model_part_name) : mpModel->CreateModelPart(r_new_model_part_name);
+
+    //TODO: RESET combined model part
+
+    this->CopyCommonData(r_combined_model_part);
```

Comment:
I think having the possibility of combining model parts from different places is very useful I will think about that. In our cases in altair everything is read inside main model part but I think it would make this modeler mucho more interesting for the rest of the community if we can read from several input mdpas.

Reply:
I was discussing this with @pooyan-dadvand and @jrubiogonzalez and we think that it might be quite tricky to do it correctly with model parts that have different origins. (Variable list should be the same, which process info should we use etc.) . So, as a first iteration, I am checking that all origin model parts have the same root.

Reply:
Perfect. I did not want you to implement the other, just to add the check :P

Reply:
done ;)

Reply:
> I was discussing this with @pooyan-dadvand and @jrubiogonzalez and we think that it might be quite tricky to do it correctly with model parts that have different origins. (Variable list should be the same, which process info should we use etc.) . So, as a first iteration, I am checking that all origin model parts have the same root.

I did not follow the discussion at all but I think that this does pretty much sense. :+1: